src/metabase/api/database.clj

```diff
@@ -551,10 +551,11 @@
              :%lower.metabase_field/name     [:like (u/lower-case-en search-string)]
```

Comment:
It might be worth commenting that an index on `lower(name)` dramatically speeds the query up in DBs that support it, but we left it for now since Maria and H2 do not support indexes on computed fields. Perhaps they'll catch up in future? I also wonder if there are other use cases for a lowercase username, we could always materialize this as another column.